src/module/linux/linux.h

```diff
@@ -1,9 +1,16 @@
 #pragma once
 // Linux specific definitions
 #include "syscalls.h"
+#include "netlink.h"
 
 #define KRF_SAFE_WRITE(x) KRF_CR0_WRITE_UNLOCK(x)
-#define KRF_LOG(...) printk(KERN_INFO __VA_ARGS__)
+#define KRF_LOG(...)                                                                               \
+  ({                                                                                               \
+    char buf[KRF_NETLINK_BUF_SIZE]; /* max message size */                                         \
+    printk(KERN_INFO __VA_ARGS__);                                                                 \
+    snprintf(buf, KRF_NETLINK_BUF_SIZE, __VA_ARGS__);                                              \
```

Comment:
We should check the return value here and handle two special cases:

* `< 0` implies a formatting error, which is unlikely but we should handle regardless
* `>= KRF_NETLINK_BUF_SIZE` implies a truncated message, since `snprintf` couldn't fit the full message into the buffer